src/components/FindMovie/FindMovie.tsx

```diff
@@ -17,41 +45,71 @@ export const FindMovie: React.FC = () => {
               id="movie-title"
               placeholder="Enter a title to search"
               className="input is-dander"
+              value={title}
+              onChange={handleTitleChange}
             />
           </div>
 
-          <p className="help is-danger" data-cy="errorMessage">
-            Can&apos;t find a movie with such a title
-          </p>
+          {
+            !isTitleCorrect
+            && (
+              <p className="help is-danger" data-cy="errorMessage">
+                Can&apos;t find a movie with such a title
+              </p>
+            )
+          }
         </div>
 
         <div className="field is-grouped">
           <div className="control">
             <button
               data-cy="searchButton"
               type="submit"
-              className="button is-light"
+              className={cn(
+                'button',
+                'is-light',
+                { 'is-loading': isMovieLoading },
+              )}
+              disabled={!title}
             >
-              Find a movie
+              {
+                !movie
+                  ? 'Find a movie'
+                  : 'Search again'
+              }
             </button>
           </div>
 
-          <div className="control">
-            <button
-              data-cy="addButton"
-              type="button"
-              className="button is-primary"
-            >
-              Add to the list
-            </button>
-          </div>
+          {
+            movie
+              && (
+                <div className="control">
+                  <button
+                    data-cy="addButton"
+                    type="button"
+                    className="button is-primary"
+                    onClick={handleAddMovie}
+                  >
+                    Add to the list
+                  </button>
+                </div>
+              )
+          }
         </div>
       </form>
 
-      <div className="container" data-cy="previewContainer">
-        <h2 className="title">Preview</h2>
-        {/* <MovieCard movie={movie} /> */}
-      </div>
+      {
+        movie
+          && (
+            <div className="container" data-cy="previewContainer">
+              <h2 className="title">Preview</h2>
+              {
+                movie
+                && <MovieCard movie={movie} />
+              }
+            </div>
+          )
```

Comment:
```suggestion
        movie
          && (
            <div className="container" data-cy="previewContainer">
              <h2 className="title">Preview</h2>
              
              <MovieCard movie={movie} />
            </div>
          )
```
You've already checked if the movie is present above. Don't duplicate yourself